apps/backend/src/modules/points-of-interest/points-of-interest.service.ts

```diff
@@ -43,6 +44,30 @@ class PointsOfInterestService implements Service {
 	}
 
 	public async delete(id: number): Promise<boolean> {
+		const poi = await this.pointsOfInterestRepository.findById(id);
+
+		if (!poi) {
+			throw new PointOfInterestError({
+				message: PointOfInterestExceptionMessage.ID_NOT_FOUND,
+				status: HTTPCode.NOT_FOUND,
+			});
+		}
+
+		const details = poi.toDetailsObject();
+
+		if (details.routes.length > 0) {
+			const [route] = details.routes;
+			const routeName = (route as { name: string }).name;
```

Comment:
It's better to reuse types for such cases instead of defining your own
```suggestion
			const { name: routeName } = route as Required<typeof details['routes'][number]>;
```